### ISSUE TITLE
ldmsd : no 'start' option

### ISSUE THREAD

**Issue by StefanA-AWS** [ovis-hpc/ovis#149]
Hi

in my running lmdsd and connecting with lmdsd_controller, there is no 'start' command, there are however 'smplr_*. This is also true for ldmsctl.

Did I miss something during the setup phase ?

Here the config :

sock:localhost:50000> version
LDMS Version: 4.1.0.0
LDMSD Version: 3.2.2.0
sock:localhost:50000> help

Documented commands (type help <topic>):
========================================
EOF                     loglevel           set_route         strgp_start    
comment                 logrotate          setgroup_add      strgp_status   
config                  oneshot            setgroup_del      strgp_stop     
connect                 plugn_query        setgroup_ins      subscribe      
daemon_exit             plugn_sets         setgroup_mod      term           
daemon_status           plugn_status       setgroup_rm       udata          
env                     prdcr_add          shell             udata_regex    
export                  prdcr_del          smplr_add         updtr_add      
failover_config         prdcr_set_status   smplr_del         updtr_del      
failover_peercfg_start  prdcr_start        smplr_start       updtr_match_add
failover_peercfg_stop   prdcr_start_regex  smplr_status      updtr_match_del
failover_start          prdcr_status       smplr_stop        updtr_prdcr_add
failover_status         prdcr_stop         source            updtr_prdcr_del
failover_stop           prdcr_stop_regex   status            updtr_start    
greeting                prdcr_subscribe    strgp_add         updtr_status   
help                    publish            strgp_del         updtr_stop     
include                 quit               strgp_metric_add  usage          
list                    say                strgp_metric_del  version        
listen                  script             strgp_prdcr_add 
load                    set                strgp_prdcr_del 

Undocumented commands:
======================
example  try

sock:localhost:50000>

**Comment by StefanA-AWS** [comment]
Sorry, question was not clear : I know about the man page to lmdsd_controller, which losts all 'smplr_*' options.
What is confusing me are the examples all using 'start' which will fail for me (as expected as there is no 'start'), so did I do something wrong in the setup or are the examples old (if yes, is there more documentation) =

**Comment by narategithub** [comment]
It seems like you installed from `master` branch, which is our development branch. Please checkout `OVIS-4.3.3` tag which is the current release.

That aside, the `version`s printed in `ldmsd_controller` are misleading. They are both API versions. I think we need to address it. To get the version information, please use `ldmsd -V`.

**Comment by valleydlr** [comment]
If you could provide your LDMSD related environment configuration information as well as the nature of the commands executed  and failure messages produced it would be useful. For example I get this for a working installation:
mhyost:~> env | grep LDMS
LDMSD_PLUGIN_LIBPATH=/opt/ovis/lib64/ovis-ldms
LDMS_AUTH_FILE=/opt/ovis/etc/ldms/ldmsauth.conf
mhyost:~> env | grep ZAP
ZAP_LIBPATH=/opt/ovis/lib64/ovis-lib

1) mhyost:~> ldmsd -x sock:50000 -l ~/ldmsd.log -v DEBUG
2) mhyost:~> ps auxw | grep ldmsd  
<user>   36932  0.0  0.0  97508  2808 ?        Ssl  16:08   0:00 ldmsd -x sock:50000 -l /home/<user>/ldmsd.log -v DEBUG
3) mhyost:~> ldms_ls -h localhost -x sock -p 50000 
returned nothing as expected
4) mhyost:~> ldmsd_controller --host=localhost --port=50000 --xprt=sock -a none 
Welcome to the LDMSD control processor
sock:localhost:50000>
5) Within ldmsd_controller:
sock:localhost:50000>load name=meminfo
sock:localhost:50000>config name=meminfo producer=myhost instance=myhost/meminfo sock:localhost:50000>component_id=1 schema=meminfo 
sock:localhost:50000>start name=meminfo interval=1000000 offset=0
sock:localhost:50000>status
Name         Type         Interval     Offset       Libpath
------------ ------------ ------------ ------------ ------------
meminfo      sampler           1000000            0 /opt/ovis/lib64/ovis-ldms/libmeminfo.so
Name             Host             Port         Transport    State
---------------- ---------------- ------------ ------------ ------------
Name             Interval     Offset       Mode            State
---------------- ------------ ------------ --------------- ------------
Name             Container        Schema           Plugin           State
---------------- ---------------- ---------------- ---------------- ------------
sock:localhost:50000>quit
6) mhyost:~> ldms_ls -h localhost -x sock -p 50000 
myhost/meminfo
mhyost:~>

**Comment by valleydlr** [comment]
Oops, 
Markdown didn't like some of it. If it is confusing let me know and I will provide a cleaner example.

**Comment by StefanA-AWS** [comment]
Thanks for the help. By using the OVIS-4 branch, I got the version which works with the examples.

Here is a list of the issues I run into :
1. If you want the follow the examples, you shouldn't install the default 'master' branch but 'OVIS-4'
2. Normally I try to avoid to work as root, however for LDMS root access is needed (or sudo, I didn't test), at least in parts. At the very minimum the starting of the demon has to be done as root.
3. swig v2 did produce link errors. It worked with swig v3, and I also think it should work with swig v4, but I only tested v4 in setup/compile phase, I didn't run it.

All of this is based on my OS image being old (not my choice) and me not using root  but I think it should be documented (in the README.md ?) 

Again, thank you all for the help !
